/**
@throws  std::runtime_error  Function can throw
*/
